pages/posts/[id].tsx

```diff
@@ -0,0 +1,60 @@
+import Post from '@modules/Post/Post';
+import type { FC, ReactElement } from 'react';
+import { EStatusCode } from '@modules/shared/api/EStatusCode';
+import type { GetServerSideProps } from 'next';
+import { getSinglePost } from '../../modules/shared/api/getSinglePost';
+import type { ISeparatePost } from '../../modules/shared/types/seperatePost/ISeparatePost';
+
+const SeparatePost: FC<ISeparatePost.IProps> = ({
+  data,
+  error,
+}): ReactElement => {
+  return (
+    <div className="flex justify-center">
+      {error.length !== 0 && (
```

Comment:
this means that you will never utilize `withErrorHandler` component however you're wrapping **Post** component with it,
please let `withErrorHandler` do it's responsibility and pass all the returned data to it